src/realm/sync/noinst/server/server.cpp

```diff
@@ -4497,7 +4525,7 @@ void SyncConnection::send_next_message()
         if (!sess) {
             // No sessions were enlisted to send
             if (REALM_LIKELY(!m_is_closing))
-                return; // Nothing more to do right now
+                break; // Check to see if there are any log messages to go out
```

Comment:
we can keep this the way it is if we move sending the log messages before the for loop.